autorest/azure/sender.go

```diff
@@ -0,0 +1,189 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// DoRetryForStatusCodes is the Azure specific implementation for retring requests
+// It also tries to register the resource provider in case it is unregistered
+func DoRetryForStatusCodes(azc Client, codes ...int) autorest.SendDecorator {
+	return func(s autorest.Sender) autorest.Sender {
+		return autorest.SenderFunc(func(r *http.Request) (resp *http.Response, err error) {
+			rr := autorest.NewRetriableRequest(r)
+			for attempt := 0; attempt <= azc.RetryAttempts; attempt++ {
+				err = rr.Prepare()
+				if err != nil {
+					return resp, err
+				}
+
+				resp, err = s.Do(rr.Request())
+				if err != nil {
+					return resp, err
+				}
+
+				if resp.StatusCode == http.StatusConflict {
+					var re RequestError
+					err = autorest.Respond(
+						resp,
+						autorest.ByUnmarshallingJSON(&re),
+						autorest.ByClosing(),
+					)
+					if err != nil {
+						return resp, err
+					}
+
+					if re.ServiceError != nil && re.ServiceError.Code == "MissingSubscriptionRegistration" {
+						azc.Sender = s
```

Comment:
Presumably for non-test scenarios azc.Sender and s are identical?  It would be good to comment why you need to do this assignment.

Reply:
They are not identical, but they contain the same information